core/fetcher.go

```diff
@@ -187,30 +218,41 @@ func (f *BlockFetcher) SubscribeNewBlockEvent(ctx context.Context) (<-chan types
 		defer close(signedBlockCh)
 		defer func() { f.isListeningForBlocks = false }()
 		for {
-			resp, err := subscription.Recv()
-			if err != nil {
-				// case where the context was not canceled but still received an error
-				if ctx.Err() == nil {
-					log.Errorw("fetcher: error receiving new height", "err", err.Error())
-				}
-				return
-			}
-			withTimeout, ctxCancel := context.WithTimeout(ctx, 10*time.Second)
-			signedBlock, err := f.GetSignedBlock(withTimeout, resp.Height)
-			ctxCancel()
-			if err != nil {
-				log.Errorw("fetcher: error receiving signed block", "height", resp.Height, "err", err.Error())
-				return
-			}
 			select {
-			case signedBlockCh <- types.EventDataSignedBlock{
-				Header:       *signedBlock.Header,
-				Commit:       *signedBlock.Commit,
-				ValidatorSet: *signedBlock.ValidatorSet,
-				Data:         *signedBlock.Data,
-			}:
 			case <-ctx.Done():
 				return
+			default:
+				resp, err := subscription.Recv()
+				if err != nil {
+					// case where the context was not canceled but still received an error
+					if ctx.Err() == nil {
```

Comment:
What's the point of the `ctx.Err` check right here?

Reply:
`subscription.Recv()` is blocking, and I didn't want to overcomplicate things by adding a channel. So, if some error happens when the subscription is waiting to receive something, we check if it's just the context cancelled to avoid printing an error in that case.

Reply:
@rach-id this solution doesn't really resolve the issue you're trying to address, though. Can you make subscription.Recv() take a context and respect its cancellation in order to return? 

otherwise, we'd need to wrap that blocking call in a way that respects ctx cancellation.

Reply:
as per sync discussion, removing this part as it's unnecessary and confusing: 600b540576b5672a522d7aa502844cf9ad9780a6